lib/std/os/linux.zig

```diff
@@ -475,22 +475,22 @@ pub const O = switch (native_arch) {
 /// Set by startup code, used by `getauxval`.
 pub var elf_aux_maybe: ?[*]std.elf.Auxv = null;
 
-const extern_getauxval = switch (builtin.zig_backend) {
+/// Whether an external or internal getauxval implementation is used.
+pub const extern_getauxval = switch (builtin.zig_backend) {
     // Calling extern functions is not yet supported with these backends
     .stage2_aarch64, .stage2_arm, .stage2_riscv64, .stage2_sparc64 => false,
     else => !builtin.link_libc,
 };
 
-comptime {
-    const root = @import("root");
-    // Export this only when building executable, otherwise it is overriding
-    // the libc implementation
-    if (extern_getauxval and (builtin.output_mode == .Exe or @hasDecl(root, "main"))) {
-        @export(&getauxvalImpl, .{ .name = "getauxval", .linkage = .weak });
-    }
-}
-
 pub const getauxval = if (extern_getauxval) struct {
+    comptime {
+        const root = @import("root");
+        // Export this only when building executable, otherwise it is overriding
+        // the libc implementation
+        if (extern_getauxval and (builtin.output_mode == .Exe or @hasDecl(root, "main"))) {
```

Comment:
I think the `extern_getauxval and` in this `if` should be superfluous, because it's already implied by the `if (extern_getauxval)` around the containing `struct`.
If that branch isn't taken, the compiler should never analyze this `comptime` block inside of it.

Reply:
Good point